WDE.Conditions/ViewModels/ConditionGroupsEditorViewModel.cs

```diff
@@ -0,0 +1,231 @@
+                node.Owner.Members.Remove(node);
+                node.Owner = null;
+            }
+        }
+
+        private void AddItemToGroup(ConditionGroupsEditorData data)
+        {
+            var result = "";
+            OpenNameEditingWindow("CONDITION_", out result);
+            if (!string.IsNullOrWhiteSpace(result))
+                data.Members.Add(new ConditionGroupsEditorDataNode(result, data));
+        }
+
+        private void EditSourceItem(ConditionGroupsEditorData data)
+        {
+            var result = data.Name;
+            OpenNameEditingWindow(data.Name, out result);
+            if (!string.IsNullOrEmpty(result))
+                data.Name = result;
+        }
+
+        private void OpenNameEditingWindow(string source, out string name)
+        {
+            var vm = new ConditionGroupsInputViewModel(source);
+            name = "";
+            if (windowManager.ShowDialog(vm))
+            {
+                if (!string.IsNullOrWhiteSpace(vm.Name))
+                    name = vm.Name;
+                else
+                {
+                    messageBoxService.ShowDialog(new MessageBoxFactory<bool>().SetTitle("Error!")
+                        .SetMainInstruction($"Group name cannot be empty!")
+                        .SetIcon(MessageBoxIcon.Error)
+                        .WithOkButton(true)
+                        .Build());
+                }
+            }
+        }
+
+        private async Task SaveGroupsToFile()
+        {
+            await conditionDataProvider.SaveConditionGroups(SourceItems.Select(x => x.ToConditionGroupsJsonData()).ToList());
+            History.MarkAsSaved();
+        }
+
+        private void MakeItems()
+        {
+            SourceItems = new ObservableCollection<ConditionGroupsEditorData>();
+
+            foreach (var item in conditionDataProvider.GetConditionGroups())
+                SourceItems.Add(new ConditionGroupsEditorData(in item));
+        }
+        
+        
+        public void Dispose()
+        {
+            historyHandler.Dispose();
+        }
+
+        public string Title { get; } = "Condition Groups Editor";
+        public ICommand Undo  => undoCommand;
+        public ICommand Redo => redoCommand;
+        public ICommand Copy { get; } = AlwaysDisabledCommand.Command;
+        public ICommand Cut { get; } = AlwaysDisabledCommand.Command;
+        public ICommand Paste { get; } = AlwaysDisabledCommand.Command;
+        public ICommand Save { get; }
+        public ICommand CloseCommand { get; set; } = null;
+        public bool CanClose { get; } = true;
+        private bool isModified;
+
+        public bool IsModified
+        {
+            get => isModified;
+            set
+            {
+                isModified = value;
+                RaisePropertyChanged(nameof(IsModified));
+            }
+        }
+        public IHistoryManager History { get; }
+    }
+    
+    public class ConditionGroupsEditorData: INotifyPropertyChanged
```

Comment:
Is your IDE automatically using that spacing? Normally there should be space after class name

Reply:
Raider is weird, sometimes it put space and sometimes not